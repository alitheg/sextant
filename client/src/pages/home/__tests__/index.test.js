// Link.react.test.js
import React from 'react';
import HomePage from '..';
import {
  BrowserRouter as Router
} from "react-router-dom";
import renderer from 'react-test-renderer';
import {cleanup, render} from '@testing-library/react';

afterEach(cleanup);

describe('Home Page', () => {
  test('Matches snapshots', () => {
    let component = renderer.create(
      <Router>
        <HomePage loggedIn={false} />
      </Router>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    component = renderer.create(
      <Router>
        <HomePage loggedIn={true} />
      </Router>
    );

    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
  test('When logged out, allows the user to log in', () => {
    const { getByText } = render(
      <Router>
        <HomePage loggedIn={false} />
      </Router>
    );

    expect(getByText(/Login/i).closest('a')).toBeTruthy()
  })
})
