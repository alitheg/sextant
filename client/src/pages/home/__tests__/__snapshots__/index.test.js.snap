// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Page Matches snapshots 1`] = `
<div
  className="jumbotron"
>
  <div
    className="container"
  >
    <h1>
      Sextant: RADIUS Management
    </h1>
    <p>
      Please log in to continue
      <a
        className="btn btn-info"
        href="/login"
        onClick={[Function]}
        onKeyDown={[Function]}
      >
        Login
      </a>
    </p>
  </div>
</div>
`;

exports[`Home Page Matches snapshots 2`] = `
<div
  className="jumbotron"
>
  <div
    className="container"
  >
    <h1>
      Sextant: RADIUS Management
    </h1>
    <p>
      Welcome
    </p>
  </div>
</div>
`;
